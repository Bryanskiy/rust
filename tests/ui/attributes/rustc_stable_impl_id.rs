#![feature(export, rustc_attrs)]

struct S;

#[export]
#[rustc_stable_impl_id]
impl S {}
//~^ ERROR: stable impl id: 0

#[export]
mod m1 {
    #[rustc_stable_impl_id]
    impl crate::S {}
    //~^ ERROR: stable impl id: 1
    #[rustc_stable_impl_id]
    impl crate::S {}
    //~^ ERROR: stable impl id: 2
    mod m2 {
        #[rustc_stable_impl_id]
        impl crate::S {}
        //~^ ERROR: stable impl id: 3
    }
}

#[export]
#[rustc_stable_impl_id]
impl S {}
//~^ ERROR: stable impl id: 4

#[export]
mod m3 {
    #[rustc_stable_impl_id]
    impl crate::S {}
    //~^ ERROR: stable impl id: 5
}

fn main() {}
