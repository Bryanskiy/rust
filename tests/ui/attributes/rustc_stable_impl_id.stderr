error: stable impl id: 0
  --> $DIR/rustc_stable_impl_id.rs:7:1
   |
LL | impl S {}
   | ^^^^^^^^^

error: stable impl id: 4
  --> $DIR/rustc_stable_impl_id.rs:27:1
   |
LL | impl S {}
   | ^^^^^^^^^

error: stable impl id: 1
  --> $DIR/rustc_stable_impl_id.rs:13:5
   |
LL |     impl crate::S {}
   |     ^^^^^^^^^^^^^^^^

error: stable impl id: 2
  --> $DIR/rustc_stable_impl_id.rs:16:5
   |
LL |     impl crate::S {}
   |     ^^^^^^^^^^^^^^^^

error: stable impl id: 3
  --> $DIR/rustc_stable_impl_id.rs:20:9
   |
LL |         impl crate::S {}
   |         ^^^^^^^^^^^^^^^^

error: stable impl id: 5
  --> $DIR/rustc_stable_impl_id.rs:33:5
   |
LL |     impl crate::S {}
   |     ^^^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

